crates/rpc-client/src/pathfinder/proofs.rs

```diff
@@ -156,3 +156,112 @@ pub fn verify_proof<H: HashFunctionType>(
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct NodeHashToNodeMappingItem {
+    pub node_hash: Felt,
+    pub node: MerkleNode,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct ContractLeavesDataItem {
+    pub nonce: Felt,
+    pub class_hash: Felt,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct ContractsProof {
+    pub nodes: Vec<NodeHashToNodeMappingItem>,
+    pub contract_leaves_data: Vec<ContractLeavesDataItem>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct GlobalRoots {
+    pub contracts_tree_root: Felt,
+    pub classes_tree_root: Felt,
+    pub block_hash: Felt,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+pub struct StorageProof {
+    pub classes_proof: Vec<NodeHashToNodeMappingItem>,
+    pub contracts_proof: ContractsProof,
+    pub contracts_storage_proofs: Vec<Vec<NodeHashToNodeMappingItem>>,
+    pub global_roots: GlobalRoots,
+}
+
+pub fn convert_storage_to_pathfinder_proof(storage_proof: StorageProof) -> PathfinderProof {
+    let mut contract_proof: Vec<TrieNode> = Vec::new();
+    for node in storage_proof.contracts_proof.nodes {
+        contract_proof.push(convert_to_trie_node(&node));
+    }
+
+    let mut storage_proofs: Vec<Vec<TrieNode>> = Vec::new();
+    for nodes in &storage_proof.contracts_storage_proofs {
+        let mut trie_nodes: Vec<TrieNode> = Vec::new();
+        for node in nodes {
+            trie_nodes.push(convert_to_trie_node(node));
+        }
+        storage_proofs.push(trie_nodes);
+    }
+
+    let contract_data: Option<ContractData>;
+    if storage_proof.contracts_storage_proofs[0].is_empty() {
+        contract_data = None;
+    } else {
+        match storage_proof.contracts_storage_proofs[0][0].node {
+            MerkleNode::Binary { left, right: _ } => {
+                contract_data = if left.to_hex_string() == "0x0" {
+                    Some(ContractData {
+                        root: storage_proof.contracts_storage_proofs[0][0].node_hash,
+                        storage_proofs: vec![],
+                    })
+                } else {
+                    Some(ContractData { root: storage_proof.contracts_storage_proofs[0][0].node_hash, storage_proofs })
+                }
+            }
+            MerkleNode::Edge { child: _, path: _, length: _ } => {
+                contract_data =
+                    Some(ContractData { root: storage_proof.contracts_storage_proofs[0][0].node_hash, storage_proofs })
+            }
+        };
+    }
+
+    PathfinderProof {
+        state_commitment: Some(storage_proof.global_roots.block_hash),
+        class_commitment: Some(storage_proof.global_roots.classes_tree_root),
+        contract_proof,
+        contract_data,
+    }
+}
+
+pub fn convert_storage_to_pathfinder_class_proof(storage_proof: StorageProof) -> PathfinderClassProof {
+    let mut class_proof: Vec<TrieNode> = Vec::new();
+    for node in &storage_proof.classes_proof {
```

Comment:
Consider a more declarative approach
```suggestion
        let class_proof = storage_proof
        .classes_proof
        .iter()
        .map(convert_to_trie_node)
        .collect();
```